screeps-game-api/src/game.rs

```diff
@@ -170,12 +172,185 @@ pub mod map {
 
     /// See [http://docs.screeps.com/api/#Game.map.isRoomAvailable]
     ///
-    /// [http://docs.screeps.com/api/#Game.map.isRoomAvailable]: http://docs.screeps.com/api/#Game.map.isRoomAvailable
+    /// [http://docs.screeps.com/api/#Game.map.isRoomAvaile]: http://docs.screeps.com/api/#Game.map.isRoomAvaile
     pub fn is_room_available(room_name: &str) -> bool {
         js_unwrap!(Game.map.isRoomAvailable(@{room_name}))
     }
+
+    /// Implements `Game.map.findExit`.
+    /// 
+    /// Does not yet support callbacks.
+    pub fn find_exit(from_room: Room, to_room: Room) -> Result<Exit, ReturnCode> {
+        let code: i32 = js_unwrap!{Game.map.findExit(@{from_room.name()}, @{to_room.name()})};
+        Exit::try_from(code).map_err(|v| v.try_into().expect("find_exit: Error code not recognized."))
+    }
+
+    #[allow(unused_variables)]
+    pub fn find_route(from_room: Room, to_room: Room, route_callback: Option<impl Fn(&str, &str) -> u32>) -> !{
+        unimplemented!()
+    }
 }
 
+pub mod market {
+    use std::collections::HashMap;
+    
+    use stdweb::unstable::TryInto;
+
+    use {Room};
+    use constants::{ReturnCode, ResourceType};
+
+    pub enum OrderType {
+        Sell,
+        Buy
+    }
+
+    impl OrderType {
+        fn as_string(&self) -> String {
```

Comment:
Since we are converting to a different type, `to_string` would be more appropriate than `as_string`.

See https://rust-lang-nursery.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv.

In addition, returning `&'static str` could be better here. Though if we do that, `as_str` might be a more appropriate name after all - it would indeed be free.

Reply:
Would you rather we apply the same principle as for the resources and pass an integer instead? Again, this is solely used to call into JS, otherwise, the type is more than explicit enough.

Reply:
If you think it's sane, sure!

I'm not 100% sure this optimization is the best idea, but then again it does seem worth it without much cost (just writing JS code to do the string->variant conversion instead of Rust code). I originally added it for resource types just because I imagined I would use those a lot. If it wouldn't be too much trouble an `__order_type_num_to_str` and `__order_type_str_to_num` pair seems like it'd be beneficial to me.

Reply:
Sorry, I think that was a bit rambly/not clear.

I'd go ahead with this and apply the same principles. I'm not completely sure that it's the best idea, but I like it anyways, and I'm confident it will make the conversions faster.